www/src/pages/insights.jsx

```diff
@@ -203,7 +320,16 @@ export default function InsightsPage({ pageContext: { tab } }) {
             <ProjectsUsage />
           </Tab>
           <Tab eventKey={INSIGHTS_TABS.COMPONENTS} title="Components">
-            <ComponentsUsage />
+            <ComponentsUsage data={components} />
```

Comment:
So, running this locally, I'm seeing it have pretty degraded performance when interacting with the UI (e.g., clicking filters, changing tabs, etc.).

I think this is because we're trying to render the contents of all the tabs even for tabs that are not active, thus doing a lot of unnecessary extra work.

Could we conditionally render the tab content only if the tab is active (example below)? Doing so seems to drastically improve the performance of this page for me locally.

```jsx
<Tab eventKey={INSIGHTS_TABS.SUMMARY} title="Summary">
  {tab === INSIGHTS_TABS.SUMMARY && (
    <SummaryUsage />
  )}
</Tab>
```

Reply:
I updated to the suggestion. But now probably the new problem is the loading time of the tab change.